solidity/random-beacon/test/DKGValidator.test.ts

```diff
@@ -0,0 +1,350 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions, no-await-in-loop */
+
+import { BigNumber } from "ethers"
+import { ethers, helpers, getUnnamedAccounts, waffle } from "hardhat"
+import { expect } from "chai"
+
+import blsData from "./data/bls"
+
+import { constants } from "./fixtures"
+import { selectGroup } from "./utils/groups"
+import { signDkgResult, DkgResult, noMisbehaved } from "./utils/dkg"
+import { Operator } from "./utils/operators"
+
+import type { SortitionPool, DKGValidator } from "../typechain"
+
+const { to1e18 } = helpers.number
+
+const fixture = async () => {
+  const SortitionPool = await ethers.getContractFactory("SortitionPool")
+  const sortitionPool = (await SortitionPool.deploy(
+    constants.poolWeightDivisor
+  )) as SortitionPool
+
+  const DKGValidator = await ethers.getContractFactory("DKGValidator")
+  const dkgValidator = (await DKGValidator.deploy(
+    sortitionPool.address
+  )) as DKGValidator
+  await dkgValidator.deployed()
+
+  return {
+    sortitionPool,
+    dkgValidator,
+  }
+}
+
+describe("DKGValidator", () => {
```

Comment:
This is not testing all possible paths - we haven't had the validation code 100% covered 😬 @tomaszslabon is working on it separately.